scripts/deploy.sh

```diff
@@ -57,121 +35,24 @@ if [[ "$DEPLOY_AIO" != "yes" ]] && [[ "$DEPLOY_HARDENING" != "yes" ]]; then
-  DATA_DISK_DEVICE=$(lsblk -brndo NAME,TYPE,RO,SIZE | \
-                     awk '/d[b-z]+ disk 0/{ if ($4>m && $4>='$DATA_DISK_MIN_SIZE'){m=$4; d=$1}}; END{print d}')
-  # Only set the secondary disk device option if there is one
-  if [ -n "${DATA_DISK_DEVICE}" ]; then
-    export BOOTSTRAP_OPTS="${BOOTSTRAP_OPTS} bootstrap_host_data_disk_device=${DATA_DISK_DEVICE}"
-  fi
-  # force the deployment of haproxy for an AIO
-  export DEPLOY_HAPROXY="yes"
-  if [[ ! -d /etc/openstack_deploy/ || ! -d /etc/openstack_deploy/$RPCD_OVERRIDES ]]; then
-
-    # Adding affinity of 3 for rabbit and galera containers
-    # This variable is passed as a user variable in the following
-    # ansible-playbook command.
-    cat > /tmp/openstack_user_config_overrides.yml <<EOF
-openstack_user_config_overrides:
-  shared-infra_hosts:
-    aio1:
-      affinity:
-        galera_container: 3
-        rabbit_mq_container: 3
-      ip: 172.29.236.100
-EOF
-    pushd $OA_DIR/tests
-      ansible-playbook bootstrap-aio.yml -i test-inventory.ini -e "${BOOTSTRAP_OPTS:-""}" -e "@/tmp/openstack_user_config_overrides.yml"
-    popd
-
-    # Create env.d directory so we can add our own configs
-    # See: https://review.openstack.org/#/c/332595/
-    mkdir -p /etc/openstack_deploy/env.d
+  # execute the AIO configuration bootstrap
+  source "$(dirname "${0}")/bootstrap-aio.sh"
 
-    # move OSA variables file to AIO location.
-    mv /etc/openstack_deploy/user_variables.yml /etc/openstack_deploy/user_osa_aio_variables.yml
-    pushd ${RPCD_DIR}
-      for filename in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
-        if [[ ! -a "/${filename}" ]]; then
-          cp "${filename}" "/${filename}";
-        fi
-      done
-    popd
-    # ensure that the elasticsearch JVM heap size is limited
-    echo "elasticsearch_heap_size_mb: 1024" >> $RPCD_OVERRIDES
-    # set the kibana admin password
-    sed -i "s/kibana_password:.*/kibana_password: ${ADMIN_PASSWORD}/" $RPCD_SECRETS
-    # set the load balancer name to the host's name
-    echo "lb_name: '$(hostname)'" >> $RPCD_OVERRIDES
-    # set the notification_plan to the default for Rackspace Cloud Servers
-    echo "maas_notification_plan: npTechnicalContactsEmail" >> $RPCD_OVERRIDES
-    # set network speed for vms
-    echo "net_max_speed: 1000" >> $RPCD_OVERRIDES
-
-    # set the necessary bits for ceph
-    if [[ "$DEPLOY_CEPH" == "yes" ]]; then
-      cp -a ${RPCD_DIR}/etc/openstack_deploy/conf.d/ceph.yml.aio /etc/openstack_deploy/conf.d/ceph.yml
-
-      # In production, the OSDs will run on bare metal however in the AIO we'll put them in containers
-      # so the MONs think we have 3 OSDs on different hosts.
-      sed -i 's/is_metal: true/is_metal: false/' /etc/openstack_deploy/env.d/ceph.yml
+  # set the kibana admin password
+  sed -i "s/kibana_password:.*/kibana_password: ${ADMIN_PASSWORD}/" $RPCD_SECRETS
```

Comment:
This could be included above, replacing it from copy to config_template.
This, or I don't see the spacing correctly.